### ISSUE TITLE
[TW-659] Solaris 10 compile error: TIOCGWINSZ was not declared in this scope

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#694]
_Owen Clarke on 2011-06-08T01:13:27Z says:_ 

 I compiled taskwarrior 1.9.4 on Solaris 10 sparc using GCC3, cmake and gmake from blastwave.org and got the following error:
<pre>
Building CXX object src/CMakeFiles/task.dir/interactive.cpp.o
~/task-1.9.4/src/interactive.cpp: In member function `int Context::getWidth()':
~/task-1.9.4/src/interactive.cpp:65: error: `TIOCGWINSZ' was not declared in this scope
~/task-1.9.4/src/interactive.cpp: In member function `int Context::getHeight()':
~/task-1.9.4/src/interactive.cpp:93: error: `TIOCGWINSZ' was not declared in this scope
gmake[2]: *** [src/CMakeFiles/task.dir/interactive.cpp.o] Error 1
gmake[1]: *** [src/CMakeFiles/task.dir/all] Error 2
gmake: *** [all] Error 2
</pre>

I had to modify interactive.cpp to #include sys/termios.h to get access to TIOCGWINSZ.
sys/ioctl.h and sys/termios.h are in the SUNWhea package, the version I have installed is below.
I'm assuming that TIOCGWINSZ is defined in sys/ioctl.h on other platforms.
<pre>
$ pkginfo -l SUNWhea
   PKGINST:  SUNWhea
      NAME:  SunOS Header Files
  CATEGORY:  system
      ARCH:  sparc
   VERSION:  11.10.0,REV=2005.01.21.15.53
   BASEDIR:  /
    VENDOR:  Sun Microsystems, Inc.
      DESC:  SunOS C/C++ header files for general development of software
    PSTAMP:  on10ptchfeat20080228002316
  INSTDATE:  Feb 28 2011 11:18
   HOTLINE:  Please contact your local service provider
    STATUS:  completely installed
     FILES:     1536 installed pathnames
                  49 shared pathnames
                 113 directories
               20823 blocks used (approx)
</pre>

System:
<pre>
$ uname -a
SunOS <hostname_snipped> 5.10 Generic_127127-11 sun4u sparc SUNW,Sun-Blade-1500
</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2011-06-08T01:13:27Z
Modified: 2014-02-09T02:06:32Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-06-08T02:54:23Z says:_ 

 Thank you Owen.  This is fixed in the 2.0 branch.  You have been added to the AUTHORS file.
